src/app/services/weather.service.ts

```diff
@@ -100,6 +100,25 @@ export class WeatherService {
             this.updateLoading(false);
           }
         });
+    if (!this.loading && lat && long) {
```

Comment:
@StreamSG in testing stuff - i found the logic is duplicated, probably from a merge conflict resolution...simple deletion needed pal